app/src/processing/app/contrib/ContributionManagerDialog.java

```diff
@@ -141,19 +141,43 @@ public void makeFrame(final Editor editor) {
     tabbedPane.addTab("Updates", null, updatesContributionTab.panel, "Updates");
     tabbedPane.setMnemonicAt(3, KeyEvent.VK_5);
     tabbedPane.setUI(new SpacedTabbedPaneUI());
+    tabbedPane.setBackground(Color.WHITE);
+    tabbedPane.setOpaque(true);
+    tabbedPane.setBorder(BorderFactory.createLineBorder(Color.BLACK, 2));
     
     
     JPanel updateTabPanel = new JPanel(true);
-    numberLabel.setOpaque(true);
-    numberLabel.setBackground(Color.BLUE);
+    numberLabel.setOpaque(false);
+    numberLabel.setBorder(BorderFactory.createEmptyBorder(0, 6, 0, 0));
+    JLabel numberBackground = new JLabel(Toolkit.getLibIcon("icons/notification.png"));
     JLabel updateTabLabel = new JLabel("Update");
+    updateTabPanel.setOpaque(false);
+    updateTabPanel.setBorder(BorderFactory.createCompoundBorder(BorderFactory
+      .createMatteBorder(0, 2, 0, 0, Color.BLACK), BorderFactory
+      .createEmptyBorder(4, 4, 4, 4)));
+    tabbedPane.setTabComponentAt(4, updateTabPanel);
+    
+    JLabel tabLabels[] = new JLabel[4];
+    for(int i = 0 ; i < tabLabels.length;i++){
+      tabLabels[i] = new JLabel(tabTitles[i]);
+      tabLabels[i]
+        .setBorder(BorderFactory.createCompoundBorder(BorderFactory
```

Comment:
This is totally unreadable for anyone trying to work on/debug this code. No need to have four line statements like this with two separate ternary operator statements.